care/facility/api/viewsets/asset.py

```diff
@@ -118,6 +120,82 @@ def get_facility(self):
     def perform_create(self, serializer):
         serializer.save(facility=self.get_facility())
 
+    @extend_schema(tags=["asset_location"])
+    @action(methods=["POST"], detail=True)
+    def duty_staff(self, request, facility_external_id, external_id):
+        """
+        Endpoint for assigning staffs to asset location
+        """
+
+        asset: AssetLocation = self.get_object()
+        duty_staff = request.data.get("duty_staff")
+
+        if not duty_staff:
+            return Response(status=status.HTTP_204_NO_CONTENT)
+
+        query = AssetLocationDutyStaff.objects.filter(
+            asset_location=asset, user__id=duty_staff, deleted=False
+        )
+
+        if query.exists():
+            raise ValidationError(
+                {"duty_staff": "Staff already assigned to the location"}
+            )
+
+        user = User.objects.filter(id=duty_staff, home_facility=asset.facility)
```

Comment:
This needs to be performed first, we need to check if a use exists before doing anything else, we also need to ensure that the id is an integer or perform some validation on it.